### DISCUSSION TITLE
【keyball61】トラックボールと一部キーが動作しない

### DISCUSSION THREAD

**Discussion by shogo1229** [Yowkees/keyball#655] [Q&A]
お世話になっております、初めての自作キーボード分からないことが多いため質問させてください。
問題に関してはタイトル通り、以下２点の問題が発生しています
1. 添付画像の赤い四角で囲っている部分が動作しない
2. トラックボールが動作しない

該当キー自体は動作しないのですが、ロープロファイル側の問題になっている部分のキースイッチを外し基盤の銀色の部分をピンセットなどで通電させると入力自体はマイコン側で確認できています。またこの時通電させてから少しの間、またはPCとの接続解除までの間は問題なく問題部分のキーが動作するという状況です。この時トラックボールは依然として動作しません。
以下２点は既に試しています。
- キーソケットを新品に付け替える
- キースイッチを動作している別の部分と入れ替える

なぜこのような状況が発生しているのか分からないため、怪しい点や、試行錯誤してみたほうがいい点などアドバイスいただけますと幸いです…！

![front](https://github.com/user-attachments/assets/084b1f7a-6695-4d62-8cb9-e259a73dd246)

![back](https://github.com/user-attachments/assets/6a5023e3-eace-4de0-9fc5-2697f90bd26b)

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。状況了解しました。

まず、入力できないキーですが、下記画像の緑丸で示したダイオードがしっかり付いていないとご指摘の２キーが動作しません。
このダイオードの極性確認と再半田付けをお願いします。
![KB61R_keyfix](https://github.com/user-attachments/assets/df75c2db-5d9b-411d-bd2f-5263831aee75)

次にトラックボールですが、ミドル基板の半田付けは問題なさそうですので、光学センサーの半田付けを見直してください。また、写真を見ると読み取り基板の刺さりが浅いようにも見えますので、コンスルーへの差し込み直しで直る可能性もあります。
問題がなさそうでしたら、下記画像の同色同士の導通を確認してください。

![KB61_TBfix](https://github.com/user-attachments/assets/4b964c33-65c7-4be2-a49a-985507b27705)

以上、お手数ですがご確認お願いいたします。

**Reply by shogo1229** [reply]
ご返答ありがとうございます、キーにつきましてはご指摘いただいた部分を修正することで無事動作しました。
ただトラックボールについては差し直しをしても動作しませんでした。
そのため、ご指摘いただいた色丸部分をテスターで導通確認を行い、全て動作することを確認できました。
おそらくセンサー基盤側の問題だと思うのですが、添付画像のどこを確認すれば良いでしょうか？
お手数おかけして申し訳ございませんが、ご回答の程よろしくお願いします。
![IMG_5863](https://github.com/user-attachments/assets/6118f6bf-a21d-440d-9739-70aa8bceedb6)
![IMG_5864](https://github.com/user-attachments/assets/490bb05f-317f-42b3-bf03-9baecf0d6e84)

**Comment by Yowkees** [comment]
ご対応ありがとうございます。

導通確認が問題なかったということで、確かにトラックボール基板が怪しいですね。
写真ありがとうございます。
これを見ると、下写真の緑丸で示した端子がテンプラしているように見えます。

半田がボール型ではなく、基板のパッドに馴染んで山型になるように半田を足して形を整えてみてください。
（念のため、山型になっていないものはすべて再加熱して形を整えてみてください。

![traboardfix](https://github.com/user-attachments/assets/594f20e7-8761-4363-9c30-89ee7bedbb00)

もし再半田しても直らないようでしたら、動作確認済のトラボ基板を販売する等で問題を切り分けたいので、白銀ラボウェブショップのお問い合わせよりご一報いただければ幸いです。

以上、お手数ですがご対応お願いいたします。

**Comment by shogo1229** [comment]
お世話になっております、再半田しても治らなかったため友人のKeyball61のトラボ基盤を拝借して動作の確認を行ったところ問題なく動作しました。
そのため今回発生した問題としてはトラボ基盤の故障だと考えているのですがトラボ基盤のみの販売等はされているのでしょうか？

**Comment by Yowkees** [comment]
ご確認ありがとうございます。
はい。トラックボール基板だけでも販売しております。
配送先等をご連絡したいので、お手数ですが下記ウェブショップのお問い合わせよりご連絡いただければ幸いです。
https://shirogane-lab.net/inquiry

以上、お手数ですがよろしくお願いいたします。
